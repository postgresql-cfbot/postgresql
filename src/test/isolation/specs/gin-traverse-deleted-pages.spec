setup
{
  create table tmp (ar int[]) with (autovacuum_enabled = false);
  insert into tmp (select array[1] from generate_series(1,10000) i);
  insert into tmp values (array[1,2]);
  insert into tmp (select array[1] from generate_series(1,10000) i);
  create index tmp_idx on tmp using gin(ar);
  delete from tmp;
}

teardown
{
  DROP TABLE tmp;
}

session "s1"
setup		{ set enable_stopevents = true;
			  set max_parallel_workers_per_gather = 0; }
step "s1s"	{ select * from tmp where ar @> array[1,2]; }

session "s2"
step "s2s1"	{ select pg_stopevent_set('EntryFindPostingLeafPage',
									  '$.datname == "isolation_regression" && $.relname == "tmp_idx"'); }
step "s2v"	{ vacuum tmp; }
step "s2s2"	{ select pg_stopevent_set('ReleaseAndReadBuffer',
									  '$.datname == "isolation_regression" && $.relname == "tmp_idx"');
			  select pg_stopevent_reset('EntryFindPostingLeafPage'); }
step "s2s3"	{ select pg_stopevent_reset('ReleaseAndReadBuffer'); }

permutation "s2s1" "s1s" "s2s2" "s2v" "s2s3"
