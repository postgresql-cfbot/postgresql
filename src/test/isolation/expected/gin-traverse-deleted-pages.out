Parsed test spec with 2 sessions

starting permutation: s2s1 s1s s2s2 s2v s2s3
step s2s1: select pg_stopevent_set('EntryFindPostingLeafPage',
									  '$.datname == "isolation_regression" && $.relname == "tmp_idx"');
pg_stopevent_set

               
step s1s: select * from tmp where ar @> array[1,2]; <waiting ...>
step s2s2: select pg_stopevent_set('ReleaseAndReadBuffer',
									  '$.datname == "isolation_regression" && $.relname == "tmp_idx"');
			  select pg_stopevent_reset('EntryFindPostingLeafPage');
pg_stopevent_set

               
pg_stopevent_reset

               
step s2v: vacuum tmp;
step s2s3: select pg_stopevent_reset('ReleaseAndReadBuffer');
pg_stopevent_reset

               
step s1s: <... completed>
ar             

